crates/driver/src/infra/config/file/load.rs

```diff
@@ -264,11 +264,14 @@ pub async fn load(network: &blockchain::Network, path: &Path) -> infra::Config {
                     config.submission.retry_interval_secs,
                 ),
                 kind: match mempool {
-                    file::Mempool::Public { revert_protection } => {
-                        mempool::Kind::Public(if *revert_protection {
-                            mempool::RevertProtection::Enabled
-                        } else {
+                    file::Mempool::Public => {
+                        // If there is only one mempool - public mempool - revert protection is
+                        // disabled, otherwise driver would not even try to settle revertable
+                        // settlements
+                        mempool::Kind::Public(if config.submission.mempools.len() == 1 {
```

Comment:
nit: It's possible to configure the `public mempool` strategy multiple times. In that case this code would enable the revert protection.
Conceptually what we should rather check is if there is **some** submission strategy which is **not** the public mempool one.

Reply:
Addressed